population/population_type_metadata.go

```diff
@@ -0,0 +1,73 @@
+package population
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/clientlog"
+	dperrors "github.com/ONSdigital/dp-api-clients-go/v2/errors"
+	"github.com/ONSdigital/log.go/v2/log"
+	"github.com/pkg/errors"
+)
+
+type GetPopulationTypeMetadataInput struct {
+	AuthTokens
+	PopulationType string
+}
+
+type GetPopulationTypeMetadataResponse struct {
+	PopulationType   string `json:"population_type"`
+	DefaultDatasetID string `json:"default_dataset_id"`
+	Edition          string `json:"edition"`
+	Version          int    `json:"version"`
+}
+
+func (c *Client) GetPopulationTypeMetadata(ctx context.Context, input GetPopulationTypeMetadataInput) (GetPopulationTypeMetadataResponse, error) {
+	logData := log.Data{
+		"method": http.MethodGet,
+	}
+	uri := fmt.Sprintf("/population-types/%s/metadata", input.PopulationType)
+	vals := url.Values{}
+	req, err := c.createGetRequest(ctx, input.UserAuthToken, input.ServiceAuthToken, uri, vals)
+	if err != nil {
+		return GetPopulationTypeMetadataResponse{}, dperrors.New(
+			err,
+			dperrors.StatusCode(err),
+			logData,
+		)
+	}
+	clientlog.Do(ctx, "getting metadata", service, req.URL.String(), logData)
```

Comment:
Although fine in this context, when this library is used in an app "getting metadata" in the log output might become ambiguous, better to be more specific.
```suggestion
	clientlog.Do(ctx, "getting population type metadata", service, req.URL.String(), logData)
```